.github/workflows/ci-publish-docs-branch.yml

```diff
@@ -38,3 +92,48 @@ jobs:
             git commit -m "docs: updating generated documentation"
             git push origin docs
           fi
+
```

Comment:
Most of the additional code below looks like it would be part of a CD pipeline, not a CI pipeline.

Also note, that the `docs` branch that gets created and pushed above with the auto generated docs has an automated PR created that only after that PR is accepted will the code be in the `main` branch. Most of the code below should probably only be executed after that PR and merge has been completed.

Reply:
splitted into ci and cd pipelines, thanks for the comment 👍